app/services/user_service.rb

```diff
@@ -1,4 +1,13 @@
 class UserService
+  def self.destroy(user:)
+    user.deactivate!
+    zombie = User.create(name: I18n.t(:'user.deleted_user'),
+                         email: "deleted-user-#{rand(10**10)}@example.com")
```

Comment:
I'd reach for `SecureRandom.uuid` here first I think, I believe that's the most common 'give me something random that won't collide' mechanism in the codebase